### ISSUE TITLE
message/extract translator extracts not correct tokens

### ISSUE THREAD

**Issue by erickskrauch** [yiisoft/yii2#16828]
### What steps will reproduce the problem?
1. Set your i18n `translator` configuration param to `['t']`.
2. Create view file with the next statements:
  ```php
  <?= Yii::t('category', 'Should not find this') ?>
  <?= t('category', 'but find this') ?>
  ```
3. Run `message/extract`.

### What is the expected result?

```php
<?php
return [
    'but find this' => '',
];
```

### What do you get instead?

```php
<?php
return [
    'Should not find this' => '',
    'but find this' => '',
];
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      | 7.2.11 (Docker)
| Operating system | Alpine Linux 3.8 (Docker)

**Comment by erickskrauch** [comment]
I would like to work on this issue, but I don’t quite know how best to solve this problem. Obviously, I need to add the logic of "looking back", but how to determine exactly that there are no other call parts before? The direct and the most stupid solution is to just check that it's not a static call, but then I can break case with nested calls.

**Comment by samdark** [comment]
Which "case with nested calls"? Is it covered by tests?

**Comment by erickskrauch** [comment]
@samdark I added additional test cases and a possible solution to this issue.

**Comment by samdark** [comment]
How about `$this->t()`?

**Comment by erickskrauch** [comment]
@samdark thanks for the mention. Fixed.

**Comment by fl0v** [comment]
i think this fix has some bugs, check line [#504](https://github.com/yiisoft/yii2/commit/5208ef761ed0552d1a4144589b1b7dff4c6fe3e7#diff-c4a5086d29e4d4b9309010d8f4a58e01R504)

**Comment by samdark** [comment]
Would you please create another issue? Thanks.
